src/smartpeak/source/ui/LoadSessionWizard.cpp

```diff
@@ -29,43 +29,95 @@ namespace SmartPeak
-    filenames->setTag(Filenames::Tag::MAIN_DIR, main_dir);
+    loaded_filenames_ = LoadFilenames::loadFilenamesFromDB(filename);
+    if (!loaded_filenames_)
+    {
+      LOGE << "Failed to load input filenames list from session DB. Loading aborted.";
+      return false;
+    }
     // Popup Session Files management if some files are not existing
-    if (filenames)
+    auto main_dir = std::filesystem::path(filename).remove_filename().generic_string();
+    loaded_filenames_->setTagValue(Filenames::Tag::MAIN_DIR, main_dir);
+    missing_input_output_directories_ = missingInputOutputDirectories();
+    if (missing_input_output_directories_)
+    {
+      set_input_output_widget->open(*loaded_filenames_, this);
+    }
+    else
+    {
+      onInputOutputSet();
+    }
+    return true;
+  };
+
+  bool LoadSessionWizard::missingInputFileNames()
+  {
+    for (const auto& file_id : (*loaded_filenames_).getFileIds())
     {
-      bool missing_file = false;
-      for (const auto& file_id : (*filenames).getFileIds())
+      const auto full_path = (*loaded_filenames_).getFullPath(file_id);
+      if ((!(*loaded_filenames_).isEmbedded(file_id)) && (!full_path.empty()) && (!std::filesystem::exists(full_path)))
       {
-        const auto full_path = (*filenames).getFullPath(file_id);
-        if ((!(*filenames).isEmbedded(file_id)) && (!full_path.empty()) && (!std::filesystem::exists(full_path)))
-        {
-          missing_file = true;
-          break;
-        }
+        return true;
+        break;
       }
-      if (missing_file)
+    }
+    return false;
+  }
+
+  void LoadSessionWizard::onInputOutputSet()
+  {
+    missing_input_file_ = missingInputFileNames();
+    if (missing_input_file_)
+    {
+      session_files_widget_manage_->open(*loaded_filenames_);
+    }
+    else
+    {
+      application_handler_.closeSession();
+      application_handler_.filenames_ = *loaded_filenames_;
+      application_handler_.main_dir_ = loaded_filenames_->getTagValue(Filenames::Tag::MAIN_DIR);
+      LoadSession load_session(
+        application_handler_,
+        workflow_manager_,
+        application_processor_observer_,
+        sequence_processor_observer_,
+        sequence_segment_processor_observer_,
+        sample_group_processor_observer_);
+      load_session.addApplicationProcessorObserver(application_processor_observer_);
+      load_session.filenames_ = loaded_filenames_;
+      load_session.notifyApplicationProcessorStart({ load_session.getName() }); // we need to use the workflow manager to profit from the progressbar
+      if (!load_session.process())
       {
-        session_files_widget_manage_->open(*filenames);
+        // load failed
+        application_handler_.closeSession();
+        load_session.notifyApplicationProcessorError("Failed to load session.");
       }
-      else
+      load_session.notifyApplicationProcessorEnd();
+    }
+  }
+
+  void LoadSessionWizard::onInputOutputCancel()
+  {
+
```

Comment:
Is this intended?

Reply:
Yes, actually the LoadSessionWizard will just not continue, so it stays empty (it's not possible, one day we had to clean something). I'm not for doing a function that does nothing, but as its method is an observer method interface, it has to be overridden. and to remove it from the observer API may be also confusing.